contracts/evm/src/dvn/utils/PacketV1Codec.sol

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: LZBL-1.2
+
+pragma solidity ^0.8.20;
+
+import {Packet} from "../interfaces/ISendLib.sol";
+import {AddressCast} from "./AddressCast.sol";
+
+library PacketV1Codec {
```

Comment:
Is this from layerzero? let's not handroll decoding here. If Layerzero changes we are updating all of our contracts which is not going to be fun!